[inductor] [FX graph cache] Ignore unbacked symints in guards expression

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126251

Summary: Found a unit test that was causing an assertion failure during an attempt to use unbacked symints in the guards expression, but it turns out unbacked symints can't affect guards anyway, so we can just filter them out. Also in this diff: test_torchinductor_dynamic_shapes.py was not configured to exercise the codecache because the TestCase setUp method was indavertently skipping the setUp of the immediate parent class.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

masnesral committed:
[inductor] [FX graph cache] Ignore unbacked symints in guards expression

Summary: Found a unit test that was causing an assertion failure during an attempt to use unbacked symints in the guards expression, but it turns out unbacked symints can't affect guards anyway, so we can just filter them out. Also in this diff: test_torchinductor_dynamic_shapes.py was not configured to exercise the codecache because the TestCase setUp method was indavertently skipping the setUp of the immediate parent class.

[ghstack-poisoned]

masnesral commented on test/inductor/test_torchinductor_dynamic_shapes.py:
```diff
@@ -116,7 +116,9 @@ def setUp(self):
         if not HAS_GPU:
             self.skipTest("Triton not available")
         torch._dynamo.reset()
-        super(TestCase, self).setUp()
```
This was skipping the inductor TestCase setUp (and therefore not enabling the cache). We want to call this class's super setup; not TestCase's super setup.

peterbell10 commented on test/inductor/test_torchinductor_dynamic_shapes.py:
```diff
@@ -116,7 +116,9 @@ def setUp(self):
-        super(TestCase, self).setUp()
+        # Weird syntax because this class is used with instantiate_device_type_tests and
+        # the type of the class at runtime will be, e.g., TestInductorDynamicCPU:
+        super(type(self), self).setUp()
```
Can this not be `TestCase.setUp(self)`?

masnesral replied:
Oh yes. That looks better

masnesral committed:
Update on "[inductor] [FX graph cache] Ignore unbacked symints in guards expression"


Summary: Found a unit test that was causing an assertion failure during an attempt to use unbacked symints in the guards expression, but it turns out unbacked symints can't affect guards anyway, so we can just filter them out. Also in this diff: test_torchinductor_dynamic_shapes.py was not configured to exercise the codecache because the TestCase setUp method was indavertently skipping the setUp of the immediate parent class.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge
